src/client_side_request.cc

```diff
@@ -1518,6 +1519,17 @@ ClientHttpRequest::processRequest()
     const bool untouchedConnect = request->method == Http::METHOD_CONNECT && !redirect.status;
 
 #if USE_OPENSSL
+    auto srvBump = getConn()->serverBump();
+    // XXX: The following "if" works however the bumping step is not updated
+    // correctly by peek-and-splice code.
+    if (srvBump && srvBump->at(XactionStep::tlsBump2)) {
+        // Update request object
+        srvBump->request = request;
+        srvBump->act.step2 = sslBumpNeed_;
+        CallJobHere(85, 4, getConn(), ConnStateData, resumePeekAndSpliceStep2);
+        return;
+    }
+
     if (untouchedConnect && sslBumpNeeded()) {
```

Comment:
I did not do this change. If it is important I will try to change it however.
IF I change it the code will look something like the following:
```
if ((!srvBump  && untouchedConnect && sslBumpNeeded() )|| (srvBump && srvBump->at(XactionStep::tlsBump1))) {
...
} else if (srvBump && srvBump->at(XactionStep::tlsBump2) {
...
}
```

If we are at bumping step1, normally we have not create yet the srvBump object (this is done later by client_side.cc code ), or we have create one just to serve an error (eg an HTTP deny).

Reply:
>  IF I change it, the code will look [bad because] If we are at bumping step1, normally we have not create yet the srvBump object (this is done later by client_side.cc code ), or we have create one just to serve an error (eg an HTTP deny).

```C++
if ((!srvBump  && untouchedConnect && sslBumpNeeded() ) ||
    (srvBump && srvBump->at(XactionStep::tlsBump1))) {
    ...
    sslBumpStart();
    return;
}

if (srvBump && srvBump->at(XactionStep::tlsBump2)) {
```

Would it be OK to create the srvBump object as soon as we know that we may need to evaluate ssl_bump? It feels like there are many code places where you have to keep reminding me that srvBump may not exist in some special cases and, in those special cases, we need to checks something else to understand what is going on. Can we create srvBump in all those special cases instead?

After creating srvBump object ASAP, we can probably make sslBumpNeeded() return false when srvBump has been created already. AFAICT, sslBumpNeeded() was meant to determine _future_ need for SslBump. If we already created srvBump, then srvBump should be consulted to determine the _current_ need.

At the end, I would not be surprised if sslBumpNeed_ would be completely gone because we would just create srvBump instead of (remembering that we should create it and using sslBumpNeed_ to remember that future need). Creation of srvBump will signify the beginning of step1.

The code would then look like this:

```C++
if (untouchedConnect && srvBump && srvBump->at(XactionStep::tlsBump1)) {
    ...
    sslBumpStart();
    return;
}

if (srvBump && srvBump->at(XactionStep::tlsBump2)) {
```

The `untouchedConnect` part still looks a bit odd, but we can deal with that later, I guess.

N.B. We can rename "server" in serverBump()/etc. if needed (so that it is not specific to "server" bumping), but that is a minor naming discrepancy thing that we can do later/separately. I am a lot more worried about all the `if` statements that are very difficult to interpret correctly when nil serverBump() may mean something other than "we are not doing SslBump in any shape or form".